Python/AthleteReportForce.py

```diff
@@ -176,51 +184,39 @@ def makeTurnPlot(inputDF, turnIndices, turnSide):
 dat['LHeel_Filt'] = signal.filtfilt(b, a, dat.LHeel)
 dat['RHeel_Filt'] = signal.filtfilt(b, a, dat.RHeel)
 
-   
-RTurns = findRightTurns(dat.RTotal_Filt, dat.LTotal_Filt)
-LTurns = findLeftTurns(dat.RTotal_Filt, dat.LTotal_Filt)
+# Turn detection
+fs = 100 
+fc = 0.5
+w = fc / (fs / 2)
+b, a = signal.butter(2, w, 'low')
 
-RTurns = cleanTurns(RTurns)
-LTurns = cleanTurns(LTurns)
+Lturn_detect = signal.filtfilt(b, a, dat.LTotal)
+Rturn_detect = signal.filtfilt(b, a, dat.RTotal)
 
-makeTurnPlot(dat, LTurns, 'Left Turns')
-makeTurnPlot(dat, RTurns, 'Reft Turns')
-answer = messagebox.askyesno("Question","Is data clean?")
+Lpeaks,_ = sig.find_peaks(Lturn_detect, prominence=25)
+Rpeaks,_ = sig.find_peaks(Rturn_detect, prominence=25)
 
-if answer == False:
-    plt.close('all')
-    print('Adding file to bad file list')
+        
+# Clean up the turn detection: ensure they oscillate
+if Lpeaks[0] < Rpeaks[0]:
+    Lpeaks, Rpeaks = EnsureTurnsAlternate(Lturn_detect,Rturn_detect,Lpeaks,Rpeaks)
+
+elif Lpeaks[0] > Rpeaks[0]:
+    Rpeaks, Lpeaks = EnsureTurnsAlternate(Rturn_detect,Lturn_detect,Rpeaks,Lpeaks)
+   
 
-if answer == True:
-    plt.close('all')
-    print('Estimating point estimates')
 
 
-for i, value in enumerate(LTurns):
+for i, value in enumerate(Rpeaks):
 # Loop through all cleaned turns to calculate discrete outcome measures.
 # using right side only (left turns) first
-
-    if i < len(LTurns) - 2:
-        # index of peak turn. saved for later in case we want to use it
-        pkIdx = np.argmax(dat.RTotal_Filt[LTurns[i]:LTurns[i+1]])
-        pkIdx = value + pkIdx
-        ## Extract relevent parameters from a turn here ##
-        peakRForce.append(np.max(dat.RTotal_Filt[LTurns[i]:LTurns[i+1]]))
-    
-
+    ## Extract relevent parameters from a turn here ##
+    peakRForce.append(np.max(dat.RTotal_Filt[value-100:value+100]))
     
-for i, value in enumerate(RTurns):
+for i, value in enumerate(Lpeaks):
 # Loop through all cleaned turns to calculate discrete outcome measures.
 # using right side only (left turns) first
-
-    if i < len(RTurns) - 2:
-    # index of peak turn. saved for later in case we want to use it
-        pkIdx = np.argmax(dat.LTotal_Filt[RTurns[i]:RTurns[i+1]])
-        pkIdx = value + pkIdx
-        ## Extract relevent parameters from a turn here ##
-        peakLForce.append(np.max(dat.LTotal_Filt[RTurns[i]:RTurns[i+1]]))
-
-
+    peakLForce.append(np.max(dat.LTotal_Filt[value-100:value+100]))
```

Comment:
May be worthwhile to replace numpy functions with their nanmax or nanmean arguement